// react-mui-login-register/es/Register.js
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component, Fragment } from 'react';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import _ from 'lodash';

import ProviderChoices from "./ProviderChoices";
import LocalUserRegister from "./LocalUserRegister";

var styles = function styles(theme) {
  return {
    root: {
      "& > * + *": {
        marginTop: theme.spacing.unit * 2
      }
    },
    or: {
      textAlign: 'center'
    }
  };
};

var Register = function (_Component) {
  _inherits(Register, _Component);

  function Register() {
    _classCallCheck(this, Register);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  Register.prototype.render = function render() {
    var _props = this.props,
        classes = _props.classes,
        onRegister = _props.onRegister,
        onRegisterWithProvider = _props.onRegisterWithProvider,
        registerFailed = _props.registerFailed,
        providers = _props.providers,
        disableLocal = _props.disableLocal,
        disableRegisterProviders = _props.disableRegisterProviders;


    var showProviders = !_.isEmpty(providers) && !disableRegisterProviders;
    var showOr = !disableLocal && showProviders;

    return React.createElement(
      'div',
      { className: classes.root },
      !disableLocal && React.createElement(LocalUserRegister, { onRegister: onRegister, registerFailed: registerFailed }),
      showOr && React.createElement(
        'div',
        { className: classes.or },
        'or'
      ),
      showProviders && React.createElement(ProviderChoices, { register: true,
        onChoice: onRegisterWithProvider,
        providers: providers
      })
    );
  };

  return Register;
}(Component);

Register.propTypes = process.env.NODE_ENV !== "production" ? {
  onRegister: PropTypes.func,
  onRegisterWithProvider: PropTypes.func,
  providers: PropTypes.arrayOf(PropTypes.string),
  registerFailed: PropTypes.string,
  disableLocal: PropTypes.bool,
  disableRegisterProviders: PropTypes.bool
} : {};


export default withStyles(styles)(Register);
