// react-mui-login-register/es/components/LoginRegisterError.js
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

var styles = function styles(theme) {
  return {
    root: {
      color: theme.palette.error.main,
      fontWeight: 'bold',
      margin: [[theme.spacing.unit, 0]]
    }
  };
};

var LoginRegisterError = function (_Component) {
  _inherits(LoginRegisterError, _Component);

  function LoginRegisterError() {
    _classCallCheck(this, LoginRegisterError);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  LoginRegisterError.prototype.render = function render() {
    return React.createElement(
      'div',
      { className: this.props.classes.root },
      this.props.message
    );
  };

  return LoginRegisterError;
}(Component);

LoginRegisterError.propTypes = process.env.NODE_ENV !== "production" ? {
  message: PropTypes.string.isRequired
} : {};


export default withStyles(styles)(LoginRegisterError);
